code_generation/producer.py

```diff
@@ -118,24 +118,28 @@ def __init__(self, name, call, inputs, output, scopes, subproducers):
         self.output = output
         self.producers = subproducers
         self.scopes = scopes
+        # this can be the case for a recurive usage of a ProducerGroup, in this case, we want to use an internal quantity as output for the group
+        if output == None:
+            self.output = q.Quantity(
+                "PG_internal_quantity_%i" % self.__class__.PG_count
+            )
+            self.__class__.PG_count += 1
```

Comment:
please remove. producer groups do not necessarily have an output and we do not know whether it is picked up by an outer producer group.